test/e2e/storage/pvc_storageclass.go

```diff
@@ -18,172 +18,172 @@ package storage
+	return pv
+}
+
+func createPVC(ctx context.Context, client clientset.Interface, namespace string) *v1.PersistentVolumeClaim {
+	pvcConfig := e2epv.PersistentVolumeClaimConfig{
+		Name: "test-pvc",
+	}
+
+	ginkgo.By("Creating a PVC")
+	pvcObj := e2epv.MakePersistentVolumeClaim(pvcConfig, namespace)
+	pvc, err := e2epv.CreatePVC(ctx, client, namespace, pvcObj)
+	framework.ExpectNoError(err, "Failed to create PVC")
+	return pvc
+}
+
+func deletePVC(ctx context.Context, client clientset.Interface, namespace, pvcName string) {
+	err := client.CoreV1().PersistentVolumeClaims(namespace).Delete(ctx, pvcName, metav1.DeleteOptions{})
+	framework.ExpectNoError(err, "Failed to delete PVC %s", pvcName)
+}
+
+func createDefaultStorageClass(ctx context.Context, client clientset.Interface) *storagev1.StorageClass {
+	sc := &storagev1.StorageClass{
 		TypeMeta: metav1.TypeMeta{
 			Kind: "StorageClass",
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: prefixSC,
+			Name: "test-default-sc",
 			Annotations: map[string]string{
 				storageutil.IsDefaultStorageClassAnnotation: "true",
 			},
 		},
 		Provisioner: "fake-1",
 	}
-}
-
-func temporarilyUnsetDefaultClasses(ctx context.Context, client clientset.Interface) func() {
-	classes, err := client.StorageV1().StorageClasses().List(ctx, metav1.ListOptions{})
-	framework.ExpectNoError(err)
-
-	changedClasses := make(map[string]bool)
-
-	for _, sc := range classes.Items {
-		if sc.Annotations[storageutil.IsDefaultStorageClassAnnotation] == "true" {
-			changedClasses[sc.GetName()] = true
-			sc.Annotations[storageutil.IsDefaultStorageClassAnnotation] = "false"
-			_, err := client.StorageV1().StorageClasses().Update(ctx, &sc, metav1.UpdateOptions{})
-			framework.ExpectNoError(err)
-		}
-	}
-
-	return func() {
-		classes, err = client.StorageV1().StorageClasses().List(ctx, metav1.ListOptions{})
-		framework.ExpectNoError(err)
-		for _, sc := range classes.Items {
-			if _, found := changedClasses[sc.GetName()]; found {
-				sc.Annotations[storageutil.IsDefaultStorageClassAnnotation] = "true"
-				_, err := client.StorageV1().StorageClasses().Update(ctx, &sc, metav1.UpdateOptions{})
-				framework.ExpectNoError(err)
-			}
-		}
-	}
 
+	createdSC, err := client.StorageV1().StorageClasses().Create(ctx, sc, metav1.CreateOptions{})
+	framework.ExpectNoError(err, "Failed to create default StorageClass")
+	return createdSC
 }
 
-func waitForPVCStorageClass(ctx context.Context, c clientset.Interface, namespace, pvcName, scName string, timeout time.Duration) (*v1.PersistentVolumeClaim, error) {
-	var watchedPVC *v1.PersistentVolumeClaim
-
-	err := wait.PollUntilContextTimeout(ctx, 1*time.Second, timeout, false, func(ctx context.Context) (bool, error) {
-		var err error
-		watchedPVC, err = c.CoreV1().PersistentVolumeClaims(namespace).Get(ctx, pvcName, metav1.GetOptions{})
+func isPVCPending(ctx context.Context, client clientset.Interface, namespace, pvcName string, timeout time.Duration) bool {
```

Comment:
Do we need a custom function to check phase instead of `e2epv.WaitForPersistentVolumeClaimsPhase`? (same with `isPVCBound`)

Reply:
We could just use `e2epv.WaitForPersistentVolumeClaimsPhase` - done in latest rev.